### ISSUE TITLE
Show stats for Yoast SEO for all languages

### ISSUE THREAD

**Issue by grappler** [polylang/polylang#211]
Yoast SEO included a stats widget in the dashboard. At the moment it only shows the stats for posts in one language.

![image](https://user-images.githubusercontent.com/1785641/34808949-ad214c88-f6b3-11e7-979d-dbce6e24506a.png)

I realized this problem as I was not getting any stats for an installation and I could at first not work out why not as we had published posts in other languages but not the language that Yoast was looking at.

The code for the stats query can be found in `WPSEO_Statistics` class. I think the following code could help with a solution.

```php
add_action( 'parse_query', function( $wp_query ) {
	if ( defined( 'REST_REQUEST' ) && REST_REQUEST &&
	( $wp_query->get( 'meta_key' ) === '_yoast_wpseo_linkdex' ||
	$wp_query->get( 'meta_key' ) === '_yoast_wpseo_focuskw' ||
	$wp_query->get( 'meta_key' ) === '_yoast_wpseo_meta-robots-noindex' ) ) {
		$wp_query->set( 'lang', '' );
	}
}, 1 );
```

**Comment by Chouby** [comment]
In fact, the query is filtered by the default language because REST requests are filtered by the default language (as every requests made on frontend, if no language is specifed). Not something intentional, but I did not change anything when the REST api was introduced. I wonder if it is not time to stop filtering REST requests. See 5461a8078e2bc4d9b912cf7c4ce474221f605c9f

**Comment by grappler** [comment]
>  I wonder if it is not time to stop filtering REST requests.

I think it a good idea. You have something similar for the standard REST api with `GET /wp/v2/posts?lang=fr` and `GET /wp/v2/posts` returning the posts in all of the languages.
https://polylang.pro/doc/rest-api/

It is a better default.

**Comment by alibasheer** [comment]
This ticket fix was included in polylang 2.3, update date: 30-1-2018

When I updated polylang to 2.3, some endpoints in my REST API stopped to filtering languages.
I have this query that was getting the english categories:
`$query = get_terms( 'category', array("lang": "en") );` 

But now it is getting all categories from all languages. I reverted back polylang to version 2.2.7 to make it work again.

We should fix that, or we should provide a hook or option to make it work again.

**Comment by Chouby** [comment]
@alibasheer Until now there was no code in Polylang related to the REST API. The fact that the requests were filtered by a language (which was not clearly determined) was not intentional and clearly caused issues as this one. The behavior was also different in the free and Pro versions which was undesired.

If you want to filter the REST API requests, you have two options.
1. Write the filters yourself to match your specific need
2. Use Polylang Pro in which such filters are already implemented. For example with:
`https://yoursite.com/wp-json/wp/v2/posts?lang=en`
would return posts in English, while
`https://yoursite.com/wp-json/wp/v2/posts`
returns all posts in all languages as it is now in the free version.
